framework/include/utils/InputParameters.h

```diff
@@ -307,6 +307,18 @@ class InputParameters : public Parameters
    */
   void addCoupledVar(const std::string & name, const std::string & doc_string);
 
+  /**
+   * This method adds a deprecated coupled variable name pair.  The parser will look for variable
+   * name pair in the input file and can return a reference to the storage location
+   * for the coupled variable if found
+   * @param name The name of the coupled variable
+   * @param doc_string A description of what the coupled variable should be
+   * @param deprecation_message The deprecation message to convey to the user
+   */
+  void addDeprecatedCoupledVar(const std::string & name,
+                               const std::string & doc_string,
+                               const std::string & deprecation_message);
+
```

Comment:
It'd be a better interface to declare both name here and have only one `getParam` for the new name below that does the `isParamValid` stuff behind the scenes.

Reply:
How is that going to snake its way through the Coupleable interface?

Reply:
@dschwen let me know your thoughts on this. Otherwise this is ready for more review

Reply:
You need a lookup map from deprecated to non-deprecated name. That could happen in `Coupleable::getVarHelper`.

Reply:
You need a lookup map from deprecated to non-deprecated name. That could happen in `Coupleable::getVarHelper`.

Reply:
Coupleable has a very confusing nomenclature for what the "variable name" is (rather than the "input parameter name")...

Reply:
Anyhow, this is not really central to this PR, but such a capability would make the follow up PR (and this PR) a bit simpler. When cleaning up, only the `addDeprecatedCoupledVar` lines would have to be removed.

```
   addDeprecatedCoupledVar(old_name, new_name);
```

which would make it so that the user could still provide `old_name` (but get a deprecation message). The code itself would already be updated to use `coupledValue(new_name)` (which would internally check if a deprecation mapping exists and then look for `old_name` if `new_name` hasn't been supplied - and then printing the deprecation message - which would be automatically generated from the parameter names).